indexer/Driver.cc

```diff
@@ -599,7 +656,8 @@ class Driver {
         planner(this->options.projectRootPath), shardPaths(), compdbParser() {
     MessageQueues::deleteIfPresent(this->id, this->numWorkers());
     this->queues = MessageQueues(this->id, this->numWorkers(),
-                                 {IPC_BUFFER_MAX_SIZE, IPC_BUFFER_MAX_SIZE});
+                                 options.ipcOptions.ipcSizeHintBytes);
+    this->options.numWorkers = this->queues.driverToWorker.size();
```

Comment:
Maybe we should make the `numWorkers` field a stateful thing, and you need to access it using either `numWorkersHint()` and `numWorkers()`. The state would change from hint to "finalized" at this state.

Reply:
Let's do this in a follow-up patch...